lib/Screens/sign_up_screen.dart

```diff
@@ -0,0 +1,122 @@
+import 'package:basic_authentication_flutter_challenge/Screens/home_screen.dart';
+import 'package:flutter/material.dart';
+import 'package:basic_authentication_flutter_challenge/Services/api.dart';
+
+class SignUpScreen extends StatefulWidget {
+  const SignUpScreen({Key? key}) : super(key: key);
+
+  @override
+  SignUpScreenState createState() => SignUpScreenState();
+}
+
+class SignUpScreenState extends State<SignUpScreen> {
+  TextEditingController userNameController = TextEditingController();
+  TextEditingController passwordController = TextEditingController();
+
+  @override
+  void dispose() {
+    userNameController.dispose();
+    passwordController.dispose();
+    super.dispose();
+  }
+
+  void showSnackBar(BuildContext context, String text) {
+    final snackBar = SnackBar(
+      content: Text(text),
+      duration: const Duration(seconds: 1),
+    );
+    ScaffoldMessenger.of(context).showSnackBar(snackBar);
+  }
+
+  void homePage() {
+    Navigator.pushReplacement(
+      context,
+      MaterialPageRoute(builder: (context) => const HomePageScreen()),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+        appBar: AppBar(
+          title: const Text('login screen'),
+        ),
+        body: Column(
+          mainAxisAlignment: MainAxisAlignment.center,
+          crossAxisAlignment: CrossAxisAlignment.center,
+          // ignore: prefer_const_literals_to_create_immutables
+          children: [
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 30.0),
+              child: Form(
```

Comment:
The `Form` widget here is not being used. Either remove it or use it. You can use it by assigning a value to its `key` property. Search for how to use `Form` widget to validate `TextFormField`s